agent/token.go

```diff
@@ -35,3 +41,79 @@ func BuildAgentToken(
 
 	return at.ToJWT()
 }
+
+type WorkerTokenConfig struct {
+	Secret  string        `yaml:"jwt_secret,omitempty"`
+	Timeout time.Duration `yaml:"jwt_timeout,omitempty"`
+}
+
+var DefaultWorkerTokenConfig = WorkerTokenConfig{
+	Timeout: 60 * time.Minute,
+}
+
+type WorkerClaims struct {
+	jwt.Claims
+}
+
+type WorkerTokenProvider struct {
+	nodeID  livekit.NodeID
+	keys    [][]byte
+	timeout time.Duration
+}
+
+func NewWorkerTokenProvider(nodeID livekit.NodeID, config WorkerTokenConfig) *WorkerTokenProvider {
+	keys := bytes.Split([]byte(config.Secret), []byte(","))
+	for i := range keys {
+		keys[i] = bytes.TrimSpace(keys[i])
+	}
+
+	return &WorkerTokenProvider{
+		nodeID:  nodeID,
+		keys:    keys,
+		timeout: config.Timeout,
+	}
+}
+
+func (t *WorkerTokenProvider) Encode(claims WorkerClaims) (string, error) {
```

Comment:
don't understand the multiple keys use case, but is there a need to have an API with a key index other than 0 to encode with?

Reply:
it allows key rotation